// @jahia/react-material/js/actions/componentRendererAction.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.componentRendererAction = void 0;
var _react = _interopRequireDefault(require("react"));
var _ComponentRenderer = require("./ComponentRenderer");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var componentRendererAction = {
  init: function init(context, props) {
    var componentRenderer = props.componentRenderer;
    context.renderComponent = function (comp) {
      if (!context.componentId) {
        var id = 'actionComponent-' + context.id;
        var componentHandler = {
          id: id,
          setProps: function setProps(props) {
            return componentRenderer.setProps(id, props);
          },
          destroy: function destroy() {
            return componentRenderer.destroy(id);
          }
        };
        componentRenderer.render(id, comp);
        return componentHandler;
      }
    };
  },
  wrappers: [function (component) {
    return /*#__PURE__*/_react["default"].createElement(_ComponentRenderer.ComponentRendererConsumer, null, function (componentRenderer) {
      return /*#__PURE__*/_react["default"].cloneElement(component, {
        componentRenderer: componentRenderer
      });
    });
  }]
};
exports.componentRendererAction = componentRendererAction;
