// @jahia/react-material/esm/actions/componentRendererAction.jsx
import React from 'react';
import { ComponentRendererConsumer } from './ComponentRenderer';
var componentRendererAction = {
  init: function init(context, props) {
    var componentRenderer = props.componentRenderer;
    context.renderComponent = function (comp) {
      if (!context.componentId) {
        var id = 'actionComponent-' + context.id;
        var componentHandler = {
          id: id,
          setProps: function setProps(props) {
            return componentRenderer.setProps(id, props);
          },
          destroy: function destroy() {
            return componentRenderer.destroy(id);
          }
        };
        componentRenderer.render(id, comp);
        return componentHandler;
      }
    };
  },
  wrappers: [function (component) {
    return /*#__PURE__*/React.createElement(ComponentRendererConsumer, null, function (componentRenderer) {
      return /*#__PURE__*/React.cloneElement(component, {
        componentRenderer: componentRenderer
      });
    });
  }]
};
export { componentRendererAction };
