// @jahia/design-system-kit/esm/components/Paper/Paper.js
import _capitalize from "lodash/capitalize";
var _excluded = ["root"],
  _excluded2 = ["color", "classes"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
import { Paper as MuiPaper, withStyles } from '@material-ui/core';
import classnames from 'classnames';
var styles = function styles(theme) {
  return {
    root: {
      padding: theme.spacing.unit * 4
    },
    colorLight: {
      backgroundColor: theme.palette.ui.epsilon
    },
    colorDark: {
      backgroundColor: theme.palette.ui.gamma,
      color: theme.palette.invert.beta
    }
  };
};
var getClasses = function getClasses(_ref) {
  var color = _ref.color,
    _ref$classes = _ref.classes,
    root = _ref$classes.root,
    myClasses = _objectWithoutProperties(_ref$classes, _excluded);
  return {
    root: classnames(root, myClasses['color' + _capitalize(color)])
  };
};
var Paper = withStyles(styles, {
  name: 'DsPaper'
})(function (_ref2) {
  var color = _ref2.color,
    classes = _ref2.classes,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/React.createElement(MuiPaper, _extends({
    classes: getClasses({
      color: color,
      classes: classes
    })
  }, props));
});
Paper.propTypes = process.env.NODE_ENV !== 'production' ? {} : {};
Paper.defaultProps = {
  color: 'light'
};
Paper.displayName = 'DsPaper';
export default Paper;
