// @jahia/design-system-kit/js/layouts/page/MainLayout.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.MainLayout = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
var _TopBar = _interopRequireDefault(require("../../components/TopBar"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles(theme) {
  return {
    root: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      minHeight: 0
    },
    topBar: {
      color: theme.palette.primary.contrastText
    },
    content: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'column',
      minHeight: 0
    },
    metaNav: {
      position: 'absolute',
      width: '50%',
      height: theme.spacing.unit * 3,
      top: 0,
      right: 0,
      paddingRight: theme.spacing.unit * 4,
      textAlign: 'right',
      color: theme.palette.text.contrastText + '!important',
      background: 'linear-gradient(to right, rgba(78, 81, 86, 0) 0%, ' + theme.palette.layout.main + ' 100%) !important;',
      '& a': {
        color: 'inherit !important'
      }
    }
  };
};
var MainLayout = function MainLayout(_ref) {
  var classes = _ref.classes,
    topBarProps = _ref.topBarProps,
    help = _ref.help,
    children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement("main", {
    className: classes.root
  }, help && /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.metaNav
  }, /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    variant: "overline",
    color: "inherit"
  }, help)), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.topBar
  }, /*#__PURE__*/_react["default"].createElement(_TopBar["default"], topBarProps)), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.content
  }, children));
};
exports.MainLayout = MainLayout;
MainLayout.propTypes = {
  children: _propTypes["default"].node.isRequired,
  classes: _propTypes["default"].object.isRequired,
  topBarProps: _propTypes["default"].object.isRequired,
  help: _propTypes["default"].node
};
MainLayout.defaultProps = {
  help: null
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles))(MainLayout);
exports["default"] = _default;
