// @jahia/design-system-kit/js/components/TopBar/TopBar.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TopBar = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _Typography = require("../Typography");
var _recompose = require("recompose");
var _styleConstants = _interopRequireDefault(require("../../theme/styleConstants"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var styles = function styles(theme) {
  return {
    root: {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '0px ' + theme.spacing.unit * 3 + 'px',
      height: _styleConstants["default"].topBarHeight + 'px'
    },
    typoTitle: {
      whiteSpace: 'nowrap',
      textOverflow: 'ellipsis',
      overflow: 'hidden',
      minWidth: 0,
      maxWidth: '50vw'
    },
    head: {
      display: 'inline-block',
      verticalAlign: 'top',
      marginRight: 'auto'
    },
    topBarActions: {
      flex: '4 1 0%',
      width: 'min-content',
      minWidth: 'max-content',
      display: 'flex',
      justifyContent: 'flex-end'
    },
    topBarTitle: {
      overflow: 'hidden'
    },
    topBarContext: {
      flex: '1 1 auto',
      display: 'flex',
      flexDirection: 'column',
      overflow: 'hidden',
      minWidth: 0,
      '& button': {
        margin: '0px',
        padding: '0px'
      }
    }
  };
};
var TopBar = function TopBar(_ref) {
  var classes = _ref.classes,
    title = _ref.title,
    titleProps = _ref.titleProps,
    contextModifiers = _ref.contextModifiers,
    path = _ref.path,
    actions = _ref.actions;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root,
    "data-sel-role": "top-bar"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.topBarContext
  }, /*#__PURE__*/_react["default"].createElement("div", null, typeof path === 'string' ? /*#__PURE__*/_react["default"].createElement(_Typography.Typography, {
    noWrap: true,
    gutterBottom: true,
    variant: "omega",
    color: "invert"
  }, path) : path), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.topBarTitle
  }, /*#__PURE__*/_react["default"].createElement(_Typography.Typography, _extends({
    gutterBottom: true,
    variant: "beta",
    color: "invert",
    className: classes.typoTitle,
    "data-sel-role": "top-bar-title",
    component: "h2"
  }, titleProps), title)), /*#__PURE__*/_react["default"].createElement("div", null, contextModifiers)), actions && /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.topBarActions
  }, actions));
};
exports.TopBar = TopBar;
TopBar.propTypes = {
  actions: _propTypes["default"].object,
  classes: _propTypes["default"].object.isRequired,
  contextModifiers: _propTypes["default"].node.isRequired,
  path: _propTypes["default"].node,
  title: _propTypes["default"].string.isRequired,
  titleProps: _propTypes["default"].object
};
TopBar.defaultProps = {
  actions: null,
  path: null,
  titleProps: {}
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles))(TopBar);
exports["default"] = _default;
