// @jahia/react-material/js/input/FilterSelect.jsx
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FilterSelect = void 0;
var _flowRight2 = _interopRequireDefault(require("lodash/flowRight"));
var _find2 = _interopRequireDefault(require("lodash/find"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _reactSelect = _interopRequireWildcard(require("react-select"));
var _icons = require("@material-ui/icons");
var _core = require("@material-ui/core");
var _excluded = ["classes", "value", "options"],
  _excluded2 = ["classes", "options", "value", "onChange"];
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); Object.defineProperty(subClass, "prototype", { writable: false }); if (superClass) _setPrototypeOf(subClass, superClass); }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }
function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }
function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }
var styles = function styles() {
  return {
    root: {
      display: 'inline-block',
      minWidth: 200
    }
  };
};
var ITEM_HEIGHT = 48;
var customStyles = {
  control: function control() {
    return {
      display: 'flex',
      alignItems: 'center',
      border: 0,
      height: 'auto',
      background: 'transparent',
      '&:hover': {
        boxShadow: 'none'
      }
    };
  },
  menu: function menu() {
    return {
      backgroundColor: 'white',
      boxShadow: '1px 2px 6px #888888',
      // Should be changed as material-ui
      position: 'absolute',
      left: 0,
      top: 'calc(100% + 1px)',
      width: '100%',
      zIndex: 2,
      maxHeight: ITEM_HEIGHT * 4.5
    };
  },
  dropdownIndicator: function dropdownIndicator() {
    return {
      padding: 0
    };
  },
  clearIndicator: function clearIndicator() {
    return {
      padding: 0
    };
  },
  menuList: function menuList() {
    return {
      maxHeight: ITEM_HEIGHT * 4.5,
      overflowY: 'auto'
    };
  }
};
var Option = /*#__PURE__*/function (_React$Component) {
  _inherits(Option, _React$Component);
  var _super = _createSuper(Option);
  function Option() {
    _classCallCheck(this, Option);
    return _super.apply(this, arguments);
  }
  _createClass(Option, [{
    key: "render",
    value: function render() {
      var _this = this;
      var _this$props = this.props,
        data = _this$props.data,
        children = _this$props.children,
        isFocused = _this$props.isFocused,
        isSelected = _this$props.isSelected,
        onFocus = _this$props.onFocus;
      return /*#__PURE__*/_react["default"].createElement(_core.MenuItem, {
        selected: isFocused,
        component: "div",
        style: {
          fontWeight: isSelected ? 500 : 400
        },
        title: data.title,
        onFocus: onFocus,
        onClick: function onClick(event) {
          _this.props.selectOption(_this.props.data, event);
        }
      }, data.icon !== null && /*#__PURE__*/_react["default"].createElement(_core.ListItemIcon, null, /*#__PURE__*/_react["default"].createElement("img", {
        src: data.icon + '.png'
      })), /*#__PURE__*/_react["default"].createElement(_core.ListItemText, null, children));
    }
  }]);
  return Option;
}(_react["default"].Component);
Option.defaultProps = {
  children: null,
  isFocused: false,
  isSelected: false
};
Option.propTypes = {
  children: _propTypes["default"].element,
  isFocused: _propTypes["default"].bool,
  isSelected: _propTypes["default"].bool,
  data: _propTypes["default"].object.isRequired,
  onFocus: _propTypes["default"].func.isRequired,
  selectOption: _propTypes["default"].func.isRequired
};
var DropdownIndicator = /*#__PURE__*/function (_React$Component2) {
  _inherits(DropdownIndicator, _React$Component2);
  var _super2 = _createSuper(DropdownIndicator);
  function DropdownIndicator() {
    _classCallCheck(this, DropdownIndicator);
    return _super2.apply(this, arguments);
  }
  _createClass(DropdownIndicator, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react["default"].createElement(_reactSelect.components.DropdownIndicator, this.props, /*#__PURE__*/_react["default"].createElement(_icons.ArrowDropDown, null));
    }
  }]);
  return DropdownIndicator;
}(_react["default"].Component);
var SelectWrapped = /*#__PURE__*/function (_React$Component3) {
  _inherits(SelectWrapped, _React$Component3);
  var _super3 = _createSuper(SelectWrapped);
  function SelectWrapped() {
    _classCallCheck(this, SelectWrapped);
    return _super3.apply(this, arguments);
  }
  _createClass(SelectWrapped, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
        classes = _this$props2.classes,
        value = _this$props2.value,
        options = _this$props2.options,
        other = _objectWithoutProperties(_this$props2, _excluded);
      var optionValue = (0, _find2["default"])(options, function (data) {
        return data.value === value;
      });
      return /*#__PURE__*/_react["default"].createElement(_reactSelect["default"], _extends({
        isClearable: true,
        components: {
          Option: Option,
          DropdownIndicator: DropdownIndicator,
          IndicatorSeparator: function IndicatorSeparator() {
            return false;
          }
        },
        styles: customStyles,
        options: options,
        value: optionValue
      }, other));
    }
  }]);
  return SelectWrapped;
}(_react["default"].Component);
SelectWrapped.defaultProps = {
  options: []
};
SelectWrapped.propTypes = {
  classes: _propTypes["default"].object.isRequired,
  value: _propTypes["default"].string.isRequired,
  options: _propTypes["default"].array
};
var FilterSelectCmp = /*#__PURE__*/function (_React$Component4) {
  _inherits(FilterSelectCmp, _React$Component4);
  var _super4 = _createSuper(FilterSelectCmp);
  function FilterSelectCmp(props) {
    var _this2;
    _classCallCheck(this, FilterSelectCmp);
    _this2 = _super4.call(this, props);
    _this2.handleChange = _this2.handleChange.bind(_assertThisInitialized(_this2));
    return _this2;
  }
  _createClass(FilterSelectCmp, [{
    key: "handleChange",
    value: function handleChange(data) {
      var newValue = null;
      if (data !== null) {
        newValue = data.value;
      }
      if (this.props.onChange !== undefined) {
        this.props.onChange(newValue);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props3 = this.props,
        classes = _this$props3.classes,
        options = _this$props3.options,
        value = _this$props3.value,
        onChange = _this$props3.onChange,
        other = _objectWithoutProperties(_this$props3, _excluded2);
      return /*#__PURE__*/_react["default"].createElement(_core.Input, {
        fullWidth: true,
        classes: classes,
        inputComponent: SelectWrapped,
        value: value,
        inputProps: _objectSpread({
          options: options
        }, other),
        onChange: this.handleChange
      });
    }
  }]);
  return FilterSelectCmp;
}(_react["default"].Component);
FilterSelectCmp.defaultProps = {
  value: null,
  onChange: null
};
FilterSelectCmp.propTypes = {
  classes: _propTypes["default"].object.isRequired,
  options: _propTypes["default"].array.isRequired,
  value: _propTypes["default"].string,
  onChange: _propTypes["default"].func
};
var FilterSelect = (0, _flowRight2["default"])((0, _core.withStyles)(styles))(FilterSelectCmp);
exports.FilterSelect = FilterSelect;
