// @jahia/react-material/js/theme/anthracite/overrides/dialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anthraciteDialogTitle = exports.anthraciteDialogContent = exports.anthraciteDialogActions = exports.anthraciteDialog = void 0;
var anthraciteDialogTitle = function anthraciteDialogTitle() {
  return {
    root: {
      paddingBottom: '6px'
    }
  };
};
exports.anthraciteDialogTitle = anthraciteDialogTitle;
var anthraciteDialog = function anthraciteDialog(theme) {
  return {
    paper: {
      backgroundColor: theme.palette.background.paper
    },
    paperFullScreen: {
      '& .footer': {
        // "background": palette.type.light ? "#363f45" : "#3b3d40",
        bottom: '-8px',
        left: '-4px',
        position: 'absolute',
        width: '100%'
      }
    }
  };
};
exports.anthraciteDialog = anthraciteDialog;
var anthraciteDialogActions = function anthraciteDialogActions() {
  return {
    root: {
      padding: '10px'
    },
    action: {
      boxShadow: 'none'
    }
  };
};
exports.anthraciteDialogActions = anthraciteDialogActions;
var anthraciteDialogContent = function anthraciteDialogContent() {
  return {
    root: {
      padding: '0 24px 0 24px'
    }
  };
};
exports.anthraciteDialogContent = anthraciteDialogContent;
