// @jahia/react-material/js/theme/anthracite/overrides/list.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anthraciteListSubheader = exports.anthraciteListItemText = exports.anthraciteListItemSecondaryAction = exports.anthraciteListItemIcon = exports.anthraciteListItem = void 0;
var anthraciteListItem = function anthraciteListItem(theme) {
  return {
    root: {
      flex: '1'
    },
    button: {
      '&[role=button]': {
        backgroundColor: theme.palette.list.text,
        boxShadow: '0px 5px 1px -4px rgba(38, 38, 38, 0.4)',
        '&:hover': {
          // Initial "backgroundColor":"#d1d7dc"
          backgroundColor: theme.palette.list.text
        }
      }
    },
    container: {
      display: 'flex'
    },
    secondaryAction: {
      flex: '1',
      position: 'relative',
      top: 0
    }
  };
};
exports.anthraciteListItem = anthraciteListItem;
var anthraciteListItemIcon = {
  root: {
    marginRight: 0
  }
};
exports.anthraciteListItemIcon = anthraciteListItemIcon;
var anthraciteListItemSecondaryAction = {
  root: {
    flex: '1',
    position: 'relative',
    top: 0,
    transform: 'none'
  }
};
exports.anthraciteListItemSecondaryAction = anthraciteListItemSecondaryAction;
var anthraciteListItemText = {
  root: {
    padding: '0 10px'
  }
};
exports.anthraciteListItemText = anthraciteListItemText;
var anthraciteListSubheader = {
  root: {
    textDecoration: 'underline'
  }
};
exports.anthraciteListSubheader = anthraciteListSubheader;
