// @jahia/react-material/js/theme/anthracite/overrides/table.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anthraciteTableRow = exports.anthraciteTablePagination = exports.anthraciteTableCell = exports.anthraciteTable = void 0;
var anthraciteTable = function anthraciteTable() {
  return {
    root: {
      overflow: 'hidden'
    }
  };
};
exports.anthraciteTable = anthraciteTable;
var anthraciteTableCell = function anthraciteTableCell(theme) {
  return {
    root: {
      borderBottom: 'none',
      padding: '8px',
      '& svg': {
        verticalAlign: 'middle'
      },
      '& button': {
        width: 'unset',
        height: 'unset'
      },
      '&:last-child': {
        paddingRight: 0
      }
    },
    head: {
      boxShadow: '2px 1px 1px 0px rgba(20, 20, 21, 0.49)',
      position: 'relative',
      background: theme.palette.background.paper
    }
  };
};
exports.anthraciteTableCell = anthraciteTableCell;
var anthraciteTableRow = function anthraciteTableRow() {
  return {
    root: {
      height: '36px'
    },
    head: {
      height: 'unset'
    }
  };
};
exports.anthraciteTableRow = anthraciteTableRow;
var anthraciteTablePagination = function anthraciteTablePagination() {
  return {
    root: {},
    select: {
      boxShadow: 'none'
    }
  };
};
exports.anthraciteTablePagination = anthraciteTablePagination;
