// @jahia/react-material/js/theme/common-base/palette.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.commonPalette = void 0;
var _core = require("@material-ui/core");
var commonPalette = {
  contrastThreshold: 3,
  tonalOffset: 0.2,
  background: {
    "default": _core.colors.grey[200],
    global: _core.colors.grey[200]
  },
  "default": {
    main: _core.colors.grey[300]
  },
  primary: {
    main: _core.colors.blueGrey[600]
  },
  secondary: {
    main: '#00a0e3'
  },
  error: {
    main: _core.colors.red[400]
  },
  confirmColor: {
    main: '#00a0e3'
  },
  publish: {
    main: _core.colors.deepOrange[500]
  },
  enabled: {
    main: _core.colors.green[400]
  },
  "delete": {
    main: _core.colors.red[600]
  },
  cancelButton: {
    main: '#676767'
  },
  publicationStatus: {
    published: {
      main: '#08D000'
    },
    modified: {
      main: '#FB9926'
    },
    notPublished: {
      main: _core.colors.grey[900]
    },
    mandatoryLanguageUnpublishable: {
      main: '#F6D62F'
    },
    liveModified: {
      main: '#F6D62F'
    },
    liveOnly: {
      main: '#F6D62F'
    },
    conflict: {
      main: '#F6D62F'
    },
    mandatoryLanguageValid: {
      main: '#F6D62F'
    },
    deleted: {
      main: '#FB9926'
    },
    markedForDeletion: {
      main: '#CC0000'
    }
  }
};
exports.commonPalette = commonPalette;
