// @jahia/moonstone/dist/components/TreeView/TreeView.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import { UncontrolledTreeView } from "./UncontrolledTreeView";
import { ControlledTreeView } from "./ControlledTreeView";
export var TreeView = function TreeView(_ref) {
  var openedItems = _ref.openedItems,
      defaultOpenedItems = _ref.defaultOpenedItems,
      onOpenItem = _ref.onOpenItem,
      onCloseItem = _ref.onCloseItem,
      data = _ref.data,
      others = _objectWithoutProperties(_ref, ["openedItems", "defaultOpenedItems", "onOpenItem", "onCloseItem", "data"]);

  // If no data render nothing
  if (!Array.isArray(data) || data.length < 1) {
    return null;
  }

  if (typeof openedItems === 'undefined') {
    return /*#__PURE__*/React.createElement(UncontrolledTreeView, _extends({
      defaultOpenedItems: defaultOpenedItems,
      data: data
    }, others));
  }

  return /*#__PURE__*/React.createElement(ControlledTreeView, _extends({
    openedItems: openedItems,
    data: data,
    onOpenItem: onOpenItem,
    onCloseItem: onCloseItem
  }, others));
};
TreeView.defaultProps = {
  onClickItem: function onClickItem() {
    return undefined;
  },
  onDoubleClickItem: function onDoubleClickItem() {
    return undefined;
  },
  onContextMenuItem: function onContextMenuItem() {
    return undefined;
  },
  openedItems: undefined,
  defaultOpenedItems: [],
  selectedItems: [],
  isReversed: false
};
TreeView.displayName = 'TreeView';
