// @jahia/design-system-kit/js/components/ToolBar/ToolBar.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ToolBar = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
var _Typography = require("../Typography");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles(theme) {
  return {
    topBar: {
      background: theme.palette.ui.gamma,
      color: theme.palette.invert.beta
    },
    topBarText: {
      fontWeight: 600,
      fontSize: '16px'
    },
    topBarButton: {
      position: 'absolute',
      right: '1.67%',
      fontStyle: 'normal',
      fontWeight: 600,
      lineHeight: '18px',
      fontSize: '14px',
      textAlign: 'right',
      color: theme.palette.invert.beta
    }
  };
};
var ToolBar = function ToolBar(_ref) {
  var classes = _ref.classes,
    contextPath = _ref.contextPath,
    label = _ref.label,
    title = _ref.title;
  return /*#__PURE__*/_react["default"].createElement(_core.AppBar, {
    position: "static"
  }, /*#__PURE__*/_react["default"].createElement(_core.Toolbar, {
    className: classes.topBar
  }, /*#__PURE__*/_react["default"].createElement(_Typography.Typography, {
    color: "invert",
    variant: "epsilon",
    className: classes.topBarText
  }, title), /*#__PURE__*/_react["default"].createElement(_core.Button, {
    className: classes.topBarButton,
    onClick: function onClick() {
      window.location.href = "".concat(contextPath, "/tools");
    }
  }, /*#__PURE__*/_react["default"].createElement(_Typography.Typography, {
    color: "invert",
    variant: "zeta"
  }, label))));
};
exports.ToolBar = ToolBar;
ToolBar.defaultProps = {
  contextPath: '',
  label: '',
  title: ''
};
ToolBar.propTypes = {
  classes: _propTypes["default"].object.isRequired,
  contextPath: _propTypes["default"].string,
  label: _propTypes["default"].string,
  title: _propTypes["default"].string
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles, {
  withTheme: true
}))(ToolBar);
exports["default"] = _default;
