// @jahia/data-helper/dist/js/fragments/getSubNodesCountFragment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getSubNodesCountFragment = exports.getEncodedSubNodesCountName = void 0;
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _graphqlTag = _interopRequireDefault(require("graphql-tag"));
var _encodeName = require("./encodeName");
var _templateObject;
var fragments = {};
var getSubNodesCountFragment = function getSubNodesCountFragment(name) {
  var encodedName = getEncodedSubNodesCountName(name);
  if (!fragments[encodedName]) {
    fragments[encodedName] = {
      applyFor: 'node',
      variables: (0, _defineProperty2["default"])({}, encodedName, '[String!]!'),
      gql: (0, _graphqlTag["default"])(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["fragment SubNodesCount_", " on JCRNode {\n                ", ": children(typesFilter: {types: $", "}) {\n                    pageInfo {\n                        totalCount\n                    }\n                }\n            }"])), encodedName, encodedName, encodedName)
    };
  }
  return {
    fragment: fragments[encodedName],
    variables: (0, _defineProperty2["default"])({}, encodedName, [name])
  };
};
exports.getSubNodesCountFragment = getSubNodesCountFragment;
var getEncodedSubNodesCountName = function getEncodedSubNodesCountName(name) {
  return 'subNodesCount_' + (0, _encodeName.encodeName)(name);
};
exports.getEncodedSubNodesCountName = getEncodedSubNodesCountName;
