// @jahia/design-system-kit/js/components/ExpansionPanelDetails/ExpansionPanelDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _core = require("@material-ui/core");
var _excluded = ["root"],
  _excluded2 = ["variant", "classes"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
/* Styles applied in the component.
* root: the style of the component itself
* attributeValue: when an attribute is set
*/
var styles = function styles(theme) {
  return {
    root: {
      display: 'flex',
      padding: theme.spacing.unit + 'px ' + theme.spacing.unit * 4 + 'px ' + theme.spacing.unit * 4 + 'px'
    }
  };
};

/*
   Set custom classes of component
 */
var getClasses = function getClasses(_ref) {
  var variant = _ref.variant,
    _ref$classes = _ref.classes,
    root = _ref$classes.root,
    dsClasses = _objectWithoutProperties(_ref$classes, _excluded);
  return {
    root: (0, _classnames["default"])(root, dsClasses[variant])
  };
};

/*
   Spread new classes into original component
 */
var ExpansionPanelDetails = (0, _core.withStyles)(styles, {
  name: 'DsExpansionPanelDetails'
})(function (_ref2) {
  var variant = _ref2.variant,
    classes = _ref2.classes,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/_react["default"].createElement(_core.ExpansionPanelDetails, _extends({
    classes: getClasses({
      variant: variant,
      classes: classes
    })
  }, props));
});

/*
  Proptype of component
 */
ExpansionPanelDetails.propTypes = process.env.NODE_ENV !== 'production' ? {
  children: _propTypes["default"].node.isRequired,
  classes: _propTypes["default"].object,
  className: _propTypes["default"].string
} : {};

/*
   Default Props
 */
ExpansionPanelDetails.defaultProps = {};
ExpansionPanelDetails.displayName = 'DsExpansionPanelDetails';
var _default = ExpansionPanelDetails;
exports["default"] = _default;
