// @jahia/design-system-kit/js/components/LeftNavigation/BurgerMenuButton.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.BurgerMenuButton = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
var _classnames = _interopRequireDefault(require("classnames"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles() {
  return {
    menuButton: {
      background: 'url(' + contextJsParameters.contextPath + '/engines/jahia-anthracite/images/logos/dx_logo_solid-white.png) center/100% no-repeat',
      width: '60px',
      height: '60px',
      backgroundSize: '100%'
    },
    menuButtonBlue: {
      background: 'url(' + contextJsParameters.contextPath + '/engines/jahia-anthracite/images/dx_logo_solid.png) center/100% no-repeat'
    }
  };
};
function openMenu() {
  var clickEvent = window.top.document.createEvent('MouseEvents');
  clickEvent.initEvent('click', true, true);
  window.top.document.getElementsByClassName('editmode-managers-menu')[0].dispatchEvent(clickEvent);
}
var BurgerMenuButton = function BurgerMenuButton(_ref) {
  var classes = _ref.classes,
    isDrawerOpen = _ref.isDrawerOpen,
    title = _ref.title;
  if (isDrawerOpen) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      className: (0, _classnames["default"])(classes.menuButton, classes.menuButtonBlue),
      "data-sel-role": "burger-menu",
      onClick: openMenu
    });
  }
  return /*#__PURE__*/_react["default"].createElement(_core.Tooltip, {
    title: title,
    placement: "right"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.menuButton),
    "data-sel-role": "burger-menu",
    onClick: openMenu
  }));
};
exports.BurgerMenuButton = BurgerMenuButton;
BurgerMenuButton.propTypes = {
  isDrawerOpen: _propTypes["default"].bool,
  classes: _propTypes["default"].object.isRequired,
  title: _propTypes["default"].string.isRequired
};
BurgerMenuButton.defaultProps = {
  isDrawerOpen: false
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles, {
  name: 'DxBurgerMenuButton'
}))(BurgerMenuButton);
exports["default"] = _default;
