// @jahia/design-system-kit/js/components/LeftNavigation/LeftMenuItem.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LeftMenuItem = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
var _reactMaterial = require("@jahia/react-material");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles(theme) {
  return {
    badgeRoot: {
      display: 'block'
    },
    badgeBadge: {
      right: 0,
      top: '-8px'
    },
    listItem: {
      margin: 0,
      height: '64px',
      display: 'block',
      width: '60px!important',
      paddingBottom: '10px!important'
    },
    typographyIconLight: {
      color: theme.palette.text.contrastText,
      fontSize: '9px',
      textTransform: 'uppercase',
      transition: 'all 0.2s ease-in 0s'
    },
    typographyIcon: {
      color: theme.palette.text.secondary,
      fontSize: '9px',
      textTransform: 'uppercase',
      transition: 'all 0.2s ease-in 0s'
    },
    colorClosed: {
      fill: theme.palette.text.contrastText,
      '& [fill="backgroundColor"]': {
        fill: theme.palette.layout.dark
      }
    },
    colorOpen: {
      fill: theme.palette.text.secondary,
      '& [fill="backgroundColor"]': {
        fill: theme.palette.background.paper
      }
    }
  };
};
var LeftMenuItem = function LeftMenuItem(_ref) {
  var classes = _ref.classes,
    drawer = _ref.drawer,
    context = _ref.context;
  var _onClick = context.onClick,
    buttonLabel = context.buttonLabel,
    buttonIcon = context.buttonIcon,
    badge = context.badge;
  var icon = (0, _reactMaterial.toIconComponent)(buttonIcon, drawer ? {
    classes: {
      root: classes.colorOpen
    }
  } : {
    classes: {
      root: classes.colorClosed
    }
  });
  var Content = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, Boolean(icon) && icon, /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    className: drawer ? classes.typographyIcon : classes.typographyIconLight,
    "data-sel-role": "left-menu-item-text"
  }, buttonLabel));
  return /*#__PURE__*/_react["default"].createElement(_core.Button, {
    className: classes.listItem,
    onClick: function onClick(e) {
      return _onClick(context, e);
    }
  }, badge ? /*#__PURE__*/_react["default"].createElement(_core.Badge, {
    badgeContent: badge,
    color: "error",
    classes: {
      root: classes.badgeRoot,
      badge: classes.badgeBadge
    },
    "data-sel-role": "badge"
  }, Content) : Content);
};
exports.LeftMenuItem = LeftMenuItem;
LeftMenuItem.propTypes = {
  context: _propTypes["default"].object.isRequired,
  // eslint-disable-next-line react/boolean-prop-naming
  drawer: _propTypes["default"].bool.isRequired,
  classes: _propTypes["default"].object.isRequired
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles, {
  withTheme: true
}))(LeftMenuItem);
exports["default"] = _default;
