// @jahia/design-system-kit/js/components/LeftNavigation/LeftNavigation.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LeftNavigation = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _classnames = _interopRequireDefault(require("classnames"));
var _reactMaterial = require("@jahia/react-material");
var _LeftMenuItem = _interopRequireDefault(require("./LeftMenuItem"));
var _recompose = require("recompose");
var _styleConstants = _interopRequireDefault(require("../../theme/styleConstants"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
var styles = function styles(theme) {
  return {
    root: {
      minWidth: _styleConstants["default"].screenMargin + _styleConstants["default"].leftNavigationWidth + 'px',
      paddingLeft: theme.spacing.unit * 4,
      display: 'flex',
      flexDirection: 'column'
    },
    rootOpenDrawer: {
      zIndex: theme.zIndex.modal,
      background: theme.palette.background.paper,
      overflow: 'visible !important',
      // Safari compatibility
      '-webkit-transform-style': 'preserve-3d' // Safari compatibility
    },

    rootClosedDrawer: {
      background: theme.palette.layout.dark,
      overflow: 'hidden'
    },
    spacer: {
      flex: '1 1 0%'
    },
    menuBurger: {
      paddingRight: '0px !important',
      paddingTop: '34px !important',
      paddingLeft: '0px !important',
      paddingBottom: '22px !important',
      boxShadow: 'none !important',
      background: 'none!important',
      transition: 'none'
    },
    drawerPaper: {
      background: theme.palette.background.paper,
      position: 'absolute',
      boxShadow: '2px 0 1px -2px rgba(0, 0, 21, 0.29)',
      left: _styleConstants["default"].screenMargin + _styleConstants["default"].leftNavigationWidth,
      width: _styleConstants["default"].leftNavigationDrawerWidth
    },
    drawerPaperClose: {
      width: 0,
      overflowX: 'hidden'
    },
    list: {},
    listBottom: {},
    drawerTree: {
      marginTop: '18px'
    }
  };
};
var LeftNavigation = function LeftNavigation(_ref) {
  var _classNames;
  var context = _ref.context,
    classes = _ref.classes,
    actionsTarget = _ref.actionsTarget,
    secondaryActionsTarget = _ref.secondaryActionsTarget,
    drawer = _ref.drawer;
  var actionContext = _objectSpread(_objectSpread({}, context), {}, {
    drawer: drawer
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(classes.root, (_classNames = {}, _defineProperty(_classNames, classes.rootOpenDrawer, drawer.drawerOpen), _defineProperty(_classNames, classes.rootClosedDrawer, !drawer.drawerOpen), _classNames))
  }, /*#__PURE__*/_react["default"].createElement(_core.List, {
    className: classes.list,
    component: "nav"
  }, /*#__PURE__*/_react["default"].createElement(_reactMaterial.DisplayActions, {
    target: actionsTarget,
    context: actionContext,
    render: function render(_ref2) {
      var context = _ref2.context;
      return /*#__PURE__*/_react["default"].createElement(_LeftMenuItem["default"], {
        context: context,
        drawer: drawer.drawerOpen
      });
    }
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.spacer
  }), /*#__PURE__*/_react["default"].createElement(_core.List, {
    className: classes.listBottom,
    component: "nav"
  }, /*#__PURE__*/_react["default"].createElement(_reactMaterial.DisplayActions, {
    target: secondaryActionsTarget,
    context: actionContext,
    render: function render(_ref3) {
      var context = _ref3.context;
      return /*#__PURE__*/_react["default"].createElement(_LeftMenuItem["default"], {
        context: context,
        drawer: drawer.drawerOpen
      });
    }
  })));
};
exports.LeftNavigation = LeftNavigation;
LeftNavigation.propTypes = {
  actionsTarget: _propTypes["default"].string.isRequired,
  secondaryActionsTarget: _propTypes["default"].string.isRequired,
  context: _propTypes["default"].object.isRequired,
  drawer: _propTypes["default"].object.isRequired,
  classes: _propTypes["default"].object.isRequired
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles, {
  withTheme: true
}))(LeftNavigation);
exports["default"] = _default;
