// redux-test-utils/dist/utils/DispatchMock.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createMockDispatch = undefined;

var _fastDeepEqual = require('fast-deep-equal');

var _fastDeepEqual2 = _interopRequireDefault(_fastDeepEqual);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var createMockDispatch = exports.createMockDispatch = function createMockDispatch() {
  var actions = [];
  return {
    dispatch: function dispatch(action) {
      actions.push(action);
    },
    getActions: function getActions() {
      return actions;
    },
    getAction: function getAction(type) {
      for (var i = 0; i < actions.length; i += 1) {
        if (actions[i].type === type) {
          return actions[i];
        }
      }
      return undefined;
    },
    isActionTypeDispatched: function isActionTypeDispatched(type) {
      for (var i = 0; i < actions.length; i += 1) {
        if (actions[i].type === type) {
          return true;
        }
      }
      return false;
    },
    isActionDispatched: function isActionDispatched(action) {
      for (var i = 0; i < actions.length; i += 1) {
        if (actions[i].type === action.type) {
          if ((0, _fastDeepEqual2.default)(actions[i], action)) {
            return true;
          }
        }
      }
      return false;
    }
  };
}; /* eslint-disable import/prefer-default-export */
