// @jahia/design-system-kit/esm/layouts/page/MainLayout.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Typography, withStyles } from '@material-ui/core';
import { compose } from 'recompose';
import TopBar from '../../components/TopBar';
var styles = function styles(theme) {
  return {
    root: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'column',
      width: '100%',
      minHeight: 0
    },
    topBar: {
      color: theme.palette.primary.contrastText
    },
    content: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'column',
      minHeight: 0
    },
    metaNav: {
      position: 'absolute',
      width: '50%',
      height: theme.spacing.unit * 3,
      top: 0,
      right: 0,
      paddingRight: theme.spacing.unit * 4,
      textAlign: 'right',
      color: theme.palette.text.contrastText + '!important',
      background: 'linear-gradient(to right, rgba(78, 81, 86, 0) 0%, ' + theme.palette.layout.main + ' 100%) !important;',
      '& a': {
        color: 'inherit !important'
      }
    }
  };
};
export var MainLayout = function MainLayout(_ref) {
  var classes = _ref.classes,
    topBarProps = _ref.topBarProps,
    help = _ref.help,
    children = _ref.children;
  return /*#__PURE__*/React.createElement("main", {
    className: classes.root
  }, help && /*#__PURE__*/React.createElement("div", {
    className: classes.metaNav
  }, /*#__PURE__*/React.createElement(Typography, {
    variant: "overline",
    color: "inherit"
  }, help)), /*#__PURE__*/React.createElement("div", {
    className: classes.topBar
  }, /*#__PURE__*/React.createElement(TopBar, topBarProps)), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, children));
};
MainLayout.propTypes = {
  children: PropTypes.node.isRequired,
  classes: PropTypes.object.isRequired,
  topBarProps: PropTypes.object.isRequired,
  help: PropTypes.node
};
MainLayout.defaultProps = {
  help: null
};
export default compose(withStyles(styles))(MainLayout);
