// @jahia/design-system-kit/js/theme/default/dialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.dsDialogTitle = exports.dsDialogContent = exports.dsDialogActions = exports.dsDialog = void 0;
var dsDialogTitle = function dsDialogTitle() {
  return {
    root: {
      paddingBottom: '6px'
    }
  };
};
exports.dsDialogTitle = dsDialogTitle;
var dsDialog = function dsDialog(theme) {
  return {
    paper: {
      backgroundColor: theme.palette.background.paper
    },
    paperFullScreen: {
      '& .footer': {
        // "background": palette.type.light ? "#363f45" : "#3b3d40",
        bottom: '-8px',
        left: '-4px',
        position: 'absolute',
        width: '100%'
      }
    }
  };
};
exports.dsDialog = dsDialog;
var dsDialogActions = function dsDialogActions() {
  return {
    root: {
      padding: '10px'
    },
    action: {
      boxShadow: 'none'
    }
  };
};
exports.dsDialogActions = dsDialogActions;
var dsDialogContent = function dsDialogContent() {
  return {
    root: {
      padding: '0 24px 0 24px'
    }
  };
};
exports.dsDialogContent = dsDialogContent;
