// @jahia/design-system-kit/js/theme/default/list.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.dsListSubheader = exports.dsListItemText = exports.dsListItemSecondaryAction = exports.dsListItemIcon = exports.dsListItem = void 0;
var dsListItem = function dsListItem(theme) {
  return {
    root: {
      flex: '1'
    },
    button: {
      '&[role=button]': {
        boxShadow: 'none',
        '&:hover': {
          backgroundColor: theme.palette.background["default"]
        }
      }
    },
    container: {
      display: 'flex'
    },
    secondaryAction: {
      flex: '1',
      position: 'relative',
      top: 0
    }
  };
};
exports.dsListItem = dsListItem;
var dsListItemIcon = {
  root: {
    marginRight: 0
  }
};
exports.dsListItemIcon = dsListItemIcon;
var dsListItemSecondaryAction = {
  root: {
    flex: '1',
    position: 'relative',
    top: 0,
    transform: 'none'
  }
};
exports.dsListItemSecondaryAction = dsListItemSecondaryAction;
var dsListItemText = {
  root: {
    padding: '0 10px',
    boxShadow: 'none'
  }
};
exports.dsListItemText = dsListItemText;
var dsListSubheader = {
  root: {
    textDecoration: 'underline'
  }
};
exports.dsListSubheader = dsListSubheader;
