// @jahia/design-system-kit/js/components/Input/Input.jsx
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = void 0;
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _core = require("@material-ui/core");
var _reactNumberFormat = _interopRequireDefault(require("react-number-format"));
var _excluded = ["classes", "disabled", "error", "onBlur", "onFocus", "readOnly", "variant"],
  _excluded2 = ["readOnly", "inputDisabled", "inputAdornedStart", "inputAdornedStartFocus", "inputAdornedStartError", "inputAdornedEnd", "inputAdornedEndReadonly"],
  _excluded3 = ["value", "isInputControlled"],
  _excluded4 = ["decimalSeparator", "decimalScale", "type"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var styles = function styles(theme) {
  return {
    root: {
      borderRadius: '1px',
      background: theme.palette.ui.epsilon,
      border: "1px solid ".concat(theme.palette.ui.omega),
      boxSizing: 'border-box',
      '&:hover:not($inputDisabled):not($focused):not($error):not($readOnly)': {
        border: "1px solid ".concat(theme.palette.ui.zeta)
      },
      fontSize: theme.typography.iota.fontSize,
      transitionDuration: '.3s',
      padding: '3px 0px 3px 12px'
    },
    focused: {
      border: "1px solid ".concat(theme.palette.brand.beta)
    },
    readOnly: {
      background: theme.palette.ui.alpha,
      border: "1px solid ".concat(theme.palette.ui.alpha)
    },
    // Hack for disabled style as the default disabled style applies on both container AND input element.
    inputDisabled: {
      background: theme.palette.ui.epsilon,
      border: "1px solid ".concat(theme.palette.ui.zeta),
      color: theme.palette.font.gamma
    },
    error: {
      border: "1px solid ".concat(theme.palette.support.alpha)
    },
    inputAdornedStart: {
      transitionDuration: '.3s'
    },
    inputAdornedStartFocus: {
      color: theme.palette.brand.beta
    },
    inputAdornedStartError: {
      color: theme.palette.support.alpha
    },
    inputAdornedEnd: {
      marginRight: '8px',
      cursor: 'pointer'
    },
    inputAdornedEndReadonly: {
      pointerEvents: 'none !important'
    }
  };
};
var InputCmp = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var classes = _ref.classes,
    disabled = _ref.disabled,
    error = _ref.error,
    onBlur = _ref.onBlur,
    onFocus = _ref.onFocus,
    readOnly = _ref.readOnly,
    variant = _ref.variant,
    others = _objectWithoutProperties(_ref, _excluded);
  var _useState = (0, _react.useState)(false),
    _useState2 = _slicedToArray(_useState, 2),
    focus = _useState2[0],
    setFocus = _useState2[1];
  var handleFocus = function handleFocus(e) {
    onFocus(e);
    setFocus(true);
  };
  var handleBlur = function handleBlur(e) {
    onBlur(e);
    setFocus(false);
  };
  var icon = variant.icon,
    interactive = variant.interactive;
  var readOnlyClass = classes.readOnly,
    inputDisabled = classes.inputDisabled,
    inputAdornedStart = classes.inputAdornedStart,
    inputAdornedStartFocus = classes.inputAdornedStartFocus,
    inputAdornedStartError = classes.inputAdornedStartError,
    inputAdornedEnd = classes.inputAdornedEnd,
    inputAdornedEndReadonly = classes.inputAdornedEndReadonly,
    containerClasses = _objectWithoutProperties(classes, _excluded2);
  return /*#__PURE__*/_react["default"].createElement(_core.InputBase, _extends({
    ref: ref,
    classes: containerClasses,
    className: "".concat(readOnly ? readOnlyClass : '', " ").concat(disabled ? inputDisabled : ''),
    disabled: disabled,
    error: error,
    readOnly: readOnly,
    startAdornment: icon && /*#__PURE__*/_react["default"].createElement(_core.InputAdornment, {
      className: "".concat(inputAdornedStart, " ").concat(!readOnly && focus ? inputAdornedStartFocus : '', " ").concat(!readOnly && error ? inputAdornedStartError : ''),
      position: "start"
    }, icon),
    endAdornment: interactive && /*#__PURE__*/_react["default"].createElement(_core.InputAdornment, {
      className: "".concat(inputAdornedEnd, " ").concat(readOnly ? inputAdornedEndReadonly : ''),
      position: "end"
    }, interactive),
    onBlur: handleBlur,
    onFocus: handleFocus
  }, others));
});
var ValueForwardInputCmp = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var value = _ref2.value,
    isInputControlled = _ref2.isInputControlled,
    others = _objectWithoutProperties(_ref2, _excluded3);
  return /*#__PURE__*/_react["default"].createElement(InputCmp, _extends({
    ref: ref,
    value: isInputControlled ? value : undefined,
    defaultValue: !isInputControlled ? value : undefined
  }, others));
});
ValueForwardInputCmp.propTypes = {
  isInputControlled: _propTypes["default"].bool.isRequired,
  value: _propTypes["default"].string
};
ValueForwardInputCmp.defaultProps = {
  value: undefined
};
var WrappedInputCmp = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var decimalSeparator = _ref3.decimalSeparator,
    decimalScale = _ref3.decimalScale,
    type = _ref3.type,
    others = _objectWithoutProperties(_ref3, _excluded4);
  if (type && type === 'number') {
    return /*#__PURE__*/_react["default"].createElement(_reactNumberFormat["default"], _extends({
      ref: ref,
      isInputControlled: others.value !== undefined,
      customInput: ValueForwardInputCmp,
      decimalScale: decimalScale,
      decimalSeparator: decimalSeparator
    }, others));
  }
  return /*#__PURE__*/_react["default"].createElement(InputCmp, _extends({
    ref: ref,
    type: type
  }, others));
});
InputCmp.defaultProps = {
  disabled: false,
  defaultValue: undefined,
  error: false,
  fullWidth: false,
  inputProps: {},
  id: undefined,
  name: undefined,
  onBlur: function onBlur() {},
  onChange: function onChange() {},
  onFocus: function onFocus() {},
  readOnly: false,
  value: undefined,
  variant: {}
};
InputCmp.propTypes = {
  classes: _propTypes["default"].object.isRequired,
  defaultValue: _propTypes["default"].string,
  // eslint-disable-next-line react/boolean-prop-naming
  disabled: _propTypes["default"].bool,
  // eslint-disable-next-line react/boolean-prop-naming
  error: _propTypes["default"].bool,
  // eslint-disable-next-line react/boolean-prop-naming
  fullWidth: _propTypes["default"].bool,
  id: _propTypes["default"].string,
  inputProps: _propTypes["default"].object,
  name: _propTypes["default"].string,
  onBlur: _propTypes["default"].func,
  onChange: _propTypes["default"].func,
  onFocus: _propTypes["default"].func,
  // eslint-disable-next-line react/boolean-prop-naming
  readOnly: _propTypes["default"].bool,
  value: _propTypes["default"].string,
  variant: _propTypes["default"].shape({
    icon: _propTypes["default"].node,
    interactive: _propTypes["default"].node
  })
};
WrappedInputCmp.defaultProps = {
  decimalSeparator: '.',
  decimalScale: undefined,
  type: 'text'
};
WrappedInputCmp.propTypes = {
  decimalSeparator: _propTypes["default"].string,
  decimalScale: _propTypes["default"].number,
  type: _propTypes["default"].string
};
var Input = (0, _core.withStyles)(styles)(WrappedInputCmp);
exports.Input = Input;
Input.displayName = 'Input';
