// @jahia/moonstone/dist/components/Dropdown/DropdownMenu.js
import _extends from "@babel/runtime/helpers/extends";
import React from 'react';
import { Menu, MenuItem, Separator } from "./..";
export var DropdownMenu = function DropdownMenu(_ref) {
  var isDisplayed = _ref.isDisplayed,
      anchorPosition = _ref.anchorPosition,
      minWidth = _ref.minWidth,
      maxWidth = _ref.maxWidth,
      maxHeight = _ref.maxHeight,
      anchorEl = _ref.anchorEl,
      hasSearch = _ref.hasSearch,
      searchEmptyText = _ref.searchEmptyText,
      data = _ref.data,
      value = _ref.value,
      imageSize = _ref.imageSize,
      handleSelect = _ref.handleSelect,
      handleKeyPress = _ref.handleKeyPress,
      onClose = _ref.onClose;
  var isEmpty = data.length < 1;
  var isGrouped = !isEmpty && typeof data[0].options !== 'undefined'; // ---
  // Generate options
  // ---

  var dropdownOption = function dropdownOption(item) {
    return /*#__PURE__*/React.createElement(MenuItem, _extends({
      key: item.value,
      role: "option",
      iconStart: item.iconStart,
      label: item.label,
      iconEnd: item.iconEnd,
      isDisabled: item.isDisabled,
      isSelected: value === item.value,
      image: item.image,
      imageSize: imageSize,
      onClick: function onClick(e) {
        return handleSelect(e, item);
      },
      onKeyPress: function onKeyPress(e) {
        return handleKeyPress(e, item);
      }
    }, item.attributes));
  };

  var dropdownGrouped = function dropdownGrouped(children, groupLabel, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: "".concat(groupLabel, "-").concat(index),
      "data-option-type": "group"
    }, index > 0 && /*#__PURE__*/React.createElement(Separator, null), /*#__PURE__*/React.createElement(MenuItem, {
      variant: "title",
      label: groupLabel
    }), children.map(function (item) {
      return dropdownOption(item);
    }));
  };

  return /*#__PURE__*/React.createElement(Menu, {
    isDisplayed: isDisplayed,
    position: "fixed",
    anchorPosition: anchorPosition,
    minWidth: minWidth,
    maxWidth: maxWidth,
    maxHeight: maxHeight,
    anchorEl: anchorEl,
    hasSearch: hasSearch,
    searchEmptyText: searchEmptyText,
    onClose: onClose
  }, data.map(function (item, index) {
    if (isGrouped) {
      item.options.map(function (o) {
        return dropdownOption(o);
      });
      return dropdownGrouped(item.options, item.groupLabel, index);
    }

    return dropdownOption(item);
  }));
};
