// @jahia/moonstone/dist/components/Dropdown/Dropdown.js
import _extends from "@babel/runtime/helpers/extends";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React, { useState } from 'react';
import clsx from 'clsx';
import "./Dropdown.css";
import spacings from "../../tokens/spacings/spacing.json";
import { DropdownImageSizes, DropdownVariants, DropdownSizes } from "./Dropdown.types";
import { Menu, MenuItem } from "../Menu";
import { Typography } from "../Typography";
import { Separator } from "../Separator";
import { ChevronDown } from "../../icons";
export var Dropdown = function Dropdown(_ref) {
  var data = _ref.data,
      label = _ref.label,
      value = _ref.value,
      isDisabled = _ref.isDisabled,
      maxWidth = _ref.maxWidth,
      variant = _ref.variant,
      size = _ref.size,
      icon = _ref.icon,
      hasSearch = _ref.hasSearch,
      searchEmptyText = _ref.searchEmptyText,
      imageSize = _ref.imageSize,
      onChange = _ref.onChange,
      className = _ref.className,
      props = _objectWithoutProperties(_ref, ["data", "label", "value", "isDisabled", "maxWidth", "variant", "size", "icon", "hasSearch", "searchEmptyText", "imageSize", "onChange", "className"]);

  // Return nothing if `data` isn't an array
  if (!Array.isArray(data)) {
    return null;
  }

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      isOpened = _useState2[0],
      setIsOpened = _useState2[1];

  var _useState3 = useState(null),
      _useState4 = _slicedToArray(_useState3, 2),
      anchorEl = _useState4[0],
      setAnchorEl = _useState4[1];

  var _useState5 = useState(null),
      _useState6 = _slicedToArray(_useState5, 2),
      minWidth = _useState6[0],
      setMinWith = _useState6[1];

  var isEmpty = data.length < 1;
  var isGrouped = !isEmpty && typeof data[0].options !== 'undefined';
  var menuMinWidth = 80;
  var menuMaxWidth;
  var menuMaxHeight;
  var anchorPosition = {
    top: parseInt(spacings.spacingNano, 10),
    left: 0
  };

  switch (imageSize) {
    case DropdownImageSizes.Big:
      menuMaxWidth = '400px';
      menuMaxHeight = '440px';
      break;

    case DropdownImageSizes.Small:
      menuMaxWidth = '264px';
      menuMaxHeight = '320px';
      break;

    default:
      // Default menu size for the dropdown when no image size is provided
      menuMaxWidth = '250px';
      menuMaxHeight = '270px';
  } // ---
  // Functions to handle events
  // ---


  var handleOpenMenu = function handleOpenMenu(e) {
    var dropdownWidth = e.currentTarget.offsetWidth;
    setMinWith("".concat(dropdownWidth < menuMinWidth ? menuMinWidth : dropdownWidth, "px"));
    setAnchorEl(e.currentTarget);
    setIsOpened(true);
  };

  var handleSelect = function handleSelect(e, item) {
    if (item) {
      var canClose = !item.isDisabled;

      if (!item.isDisabled && item.value !== value) {
        e.stopPropagation();
        canClose = onChange(e, item);
      }

      if (canClose !== false) {
        setIsOpened(false);
      }
    }
  };

  var handleCloseMenu = function handleCloseMenu() {
    setIsOpened(false);
    setAnchorEl(null);
  };

  var handleKeyPress = function handleKeyPress(e, item) {
    if (e.key === 'Enter') {
      handleSelect(e, item);
    }
  }; // ---
  // CSS classes
  // ---


  var cssDropdown = clsx(!label && !icon ? 'flexRow_reverse' : 'flexRow_between', 'alignCenter', 'moonstone-dropdown', "moonstone-".concat(size), "moonstone-dropdown_".concat(variant), {
    'moonstone-disabled': typeof isDisabled === 'undefined' && isEmpty ? true : isDisabled,
    'moonstone-filled': value,
    'moonstone-opened': isOpened
  }); // ---
  // Generate options
  // ---

  var dropdownOption = function dropdownOption(item) {
    return /*#__PURE__*/React.createElement(MenuItem, _extends({
      key: item.value,
      role: "option",
      iconStart: item.iconStart,
      label: item.label,
      iconEnd: item.iconEnd,
      isDisabled: item.isDisabled,
      isSelected: value === item.value,
      image: item.image,
      imageSize: imageSize,
      onClick: function onClick(e) {
        return handleSelect(e, item);
      },
      onKeyPress: function onKeyPress(e) {
        return handleKeyPress(e, item);
      }
    }, item.attributes));
  };

  var dropdownGrouped = function dropdownGrouped(children, groupLabel, index) {
    return /*#__PURE__*/React.createElement("div", {
      key: "".concat(groupLabel, "-").concat(index),
      "data-option-type": "group"
    }, index > 0 && /*#__PURE__*/React.createElement(Separator, null), /*#__PURE__*/React.createElement(MenuItem, {
      variant: "title",
      label: groupLabel
    }), children.map(function (item) {
      return dropdownOption(item);
    }));
  };

  return /*#__PURE__*/React.createElement("div", _extends({
    className: clsx('moonstone-dropdown_container', className),
    style: {
      maxWidth: maxWidth
    }
  }, props, {
    onKeyPress: function onKeyPress(e) {
      if (e.key === 'Enter') {
        handleOpenMenu(e);
      }
    }
  }), /*#__PURE__*/React.createElement("div", {
    role: "dropdown",
    className: clsx(cssDropdown),
    tabIndex: 0,
    onClick: handleOpenMenu,
    onKeyPress: function onKeyPress(e) {
      if (e.key === 'Enter') {
        handleSelect(e);
      }
    }
  }, icon && /*#__PURE__*/React.createElement(icon.type, _extends({}, icon.props, {
    size: "small",
    className: clsx('moonstone-dropdown_icon')
  })), /*#__PURE__*/React.createElement(Typography, {
    isNowrap: true,
    variant: "caption",
    component: "span",
    className: clsx('flexFluid')
  }, label), /*#__PURE__*/React.createElement(ChevronDown, {
    className: "moonstone-dropdown_chevronDown"
  })), isOpened && /*#__PURE__*/React.createElement(Menu, {
    isDisplayed: true,
    position: "absolute",
    anchorPosition: anchorPosition,
    minWidth: minWidth,
    maxWidth: menuMaxWidth,
    maxHeight: menuMaxHeight,
    anchorEl: anchorEl,
    hasSearch: hasSearch,
    searchEmptyText: searchEmptyText,
    onClose: handleCloseMenu
  }, data.map(function (item, index) {
    if (isGrouped) {
      item.options.map(function (o) {
        return dropdownOption(o);
      });
      return dropdownGrouped(item.options, item.groupLabel, index);
    }

    return dropdownOption(item);
  })));
};
Dropdown.defaultProps = {
  icon: null,
  variant: DropdownVariants.Ghost,
  size: DropdownSizes.Medium,
  maxWidth: '300px',
  hasSearch: false,
  searchEmptyText: 'No results found.'
};
Dropdown.displayName = 'Dropdown';
