// @jahia/design-system-kit/js/layouts/content/FullWidthContent.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.FullWidthContent = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles() {
  return {
    root: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'column',
      minHeight: 0
    }
  };
};
var FullWidthContent = function FullWidthContent(_ref) {
  var children = _ref.children,
    classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, children);
};
exports.FullWidthContent = FullWidthContent;
FullWidthContent.propTypes = {
  children: _propTypes["default"].element.isRequired,
  classes: _propTypes["default"].object.isRequired
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles))(FullWidthContent);
exports["default"] = _default;
