// @jahia/design-system-kit/js/layouts/content/TwoColumnsContent.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TwoColumnsContent = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _recompose = require("recompose");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles(theme) {
  return {
    root: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'row',
      backgroundColor: theme.palette.ui.alpha,
      padding: theme.spacing.unit * 4 + 'px ' + theme.spacing.unit * 4 + 'px 0'
    },
    left: {
      flex: '7 1 0%',
      display: 'flex',
      flexDirection: 'column',
      paddingRight: theme.spacing.unit * 2 + 'px'
    },
    right: {
      flex: '5 1 0%',
      display: 'flex',
      flexDirection: 'column',
      paddingLeft: theme.spacing.unit * 2 + 'px'
    }
  };
};
var TwoColumnsContent = function TwoColumnsContent(_ref) {
  var children = _ref.children,
    rightCol = _ref.rightCol,
    classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.left
  }, children), /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.right
  }, rightCol));
};
exports.TwoColumnsContent = TwoColumnsContent;
TwoColumnsContent.propTypes = {
  children: _propTypes["default"].element.isRequired,
  rightCol: _propTypes["default"].element.isRequired,
  classes: _propTypes["default"].object.isRequired
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles))(TwoColumnsContent);
exports["default"] = _default;
