// @jahia/design-system-kit/js/components/PropTypesConstants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TextVariants = exports.TextColors = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var TextColors = ['inherit', 'alpha', 'beta', 'gamma', 'invert'];
exports.TextColors = TextColors;
var TextVariants = ['giga', 'mega', 'alpha', 'beta', 'gamma', 'delta', 'epsilon', 'zeta', 'iota', 'omega', 'caption', 'legal', 'p'];
exports.TextVariants = TextVariants;
var _default = {
  TextColors: _propTypes["default"].oneOf(TextColors),
  TextVariants: _propTypes["default"].oneOf(TextVariants),
  ButtonVariants: _propTypes["default"].oneOf(['ghost', 'primary', 'secondary', 'inline']),
  ButtonColors: _propTypes["default"].oneOf(['default', 'inverted']),
  ButtonSizes: _propTypes["default"].oneOf(['normal', 'compact']),
  ExpansionPanelVariants: _propTypes["default"].oneOf(['normal', 'ghost']),
  ExpansionPanelColors: _propTypes["default"].oneOf(['default', 'inverted']),
  IconButtonVariants: _propTypes["default"].oneOf(['ghost', 'normal']),
  IconButtonSizes: _propTypes["default"].oneOf(['normal', 'compact']),
  IconButtonColors: _propTypes["default"].oneOf(['default', 'inverted', 'inherit']),
  SelectColors: _propTypes["default"].oneOf(['default', 'inverted']),
  SelectVariants: _propTypes["default"].oneOf(['normal', 'ghost']),
  ChipSizes: _propTypes["default"].oneOf(['normal', 'compact']),
  ChipColors: _propTypes["default"].oneOf(['default', 'inverted']),
  ChipVariants: _propTypes["default"].oneOf(['primary', 'secondary'])
};
exports["default"] = _default;
