// @jahia/design-system-kit/js/components/LeftNavigation/LeftDrawerContent/LeftDrawerListItems.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LeftDrawerListItems = void 0;
var _lodash2 = _interopRequireDefault(require("lodash/lodash"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _core = require("@material-ui/core");
var _icons = require("@material-ui/icons");
var _react = _interopRequireDefault(require("react"));
var _reactMaterial = require("@jahia/react-material");
var _recompose = require("recompose");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
var styles = function styles(theme) {
  return {
    expand: {
      width: theme.spacing.unit * 3 + 'px'
    },
    nested: {
      paddingLeft: theme.spacing.unit
    }
  };
};
var LeftDrawerListItems = function LeftDrawerListItems(_ref) {
  var context = _ref.context,
    actionPath = _ref.actionPath,
    classes = _ref.classes;
  return /*#__PURE__*/_react["default"].createElement(_reactMaterial.DisplayActions, {
    target: context.menu,
    context: _objectSpread(_objectSpread({}, context.originalContext), {}, {
      parent: context
    }),
    render: function render(actionProps) {
      var actionContext = actionProps.context;
      var icon = actionContext.buttonIcon;
      actionContext.actionPath = actionPath + '/' + actionContext.key;
      icon = (0, _reactMaterial.toIconComponent)(icon, {
        fontSize: 'small'
      });
      return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_core.ListItem, {
        button: true,
        selected: _lodash2["default"].includes(_lodash2["default"].split(actionPath, '/'), actionContext.actionKey),
        onClick: function onClick(event) {
          if (!actionContext.hasChildren) {
            context.drawer.handleDrawerClose();
          }
          actionContext.onClick(actionContext, event);
        }
      }, /*#__PURE__*/_react["default"].createElement("div", {
        className: classes.expand
      }, actionContext.hasChildren ? actionContext.open || actionContext.selected ? /*#__PURE__*/_react["default"].createElement(_icons.ExpandMore, {
        fontSize: "small"
      }) : /*#__PURE__*/_react["default"].createElement(_icons.ChevronRight, {
        fontSize: "small"
      }) : null), icon, "\xA0", /*#__PURE__*/_react["default"].createElement(_core.Typography, {
        color: "textPrimary"
      }, actionContext.buttonLabel)), /*#__PURE__*/_react["default"].createElement(_core.List, {
        disablePadding: true,
        classes: {
          root: classes.nested
        }
      }, actionContext.menu && actionContext.open && /*#__PURE__*/_react["default"].createElement(LeftDrawerListItems, {
        context: actionContext,
        actionPath: actionPath + '/' + actionContext.key,
        classes: classes
      })));
    }
  });
};
exports.LeftDrawerListItems = LeftDrawerListItems;
LeftDrawerListItems.propTypes = {
  actionPath: _propTypes["default"].string.isRequired,
  context: _propTypes["default"].object.isRequired,
  classes: _propTypes["default"].object.isRequired
};
var _default = (0, _recompose.compose)((0, _core.withStyles)(styles))(LeftDrawerListItems);
exports["default"] = _default;
