// @jahia/react-material/esm/theme/anthracite/overrides/list.js
var anthraciteListItem = function anthraciteListItem(theme) {
  return {
    root: {
      flex: '1'
    },
    button: {
      '&[role=button]': {
        backgroundColor: theme.palette.list.text,
        boxShadow: '0px 5px 1px -4px rgba(38, 38, 38, 0.4)',
        '&:hover': {
          // Initial "backgroundColor":"#d1d7dc"
          backgroundColor: theme.palette.list.text
        }
      }
    },
    container: {
      display: 'flex'
    },
    secondaryAction: {
      flex: '1',
      position: 'relative',
      top: 0
    }
  };
};
var anthraciteListItemIcon = {
  root: {
    marginRight: 0
  }
};
var anthraciteListItemSecondaryAction = {
  root: {
    flex: '1',
    position: 'relative',
    top: 0,
    transform: 'none'
  }
};
var anthraciteListItemText = {
  root: {
    padding: '0 10px'
  }
};
var anthraciteListSubheader = {
  root: {
    textDecoration: 'underline'
  }
};
export { anthraciteListItemIcon, anthraciteListItem, anthraciteListItemSecondaryAction, anthraciteListItemText, anthraciteListSubheader };
