// @jahia/react-material/esm/theme/anthracite/overrides/table.js
var anthraciteTable = function anthraciteTable() {
  return {
    root: {
      overflow: 'hidden'
    }
  };
};
var anthraciteTableCell = function anthraciteTableCell(theme) {
  return {
    root: {
      borderBottom: 'none',
      padding: '8px',
      '& svg': {
        verticalAlign: 'middle'
      },
      '& button': {
        width: 'unset',
        height: 'unset'
      },
      '&:last-child': {
        paddingRight: 0
      }
    },
    head: {
      boxShadow: '2px 1px 1px 0px rgba(20, 20, 21, 0.49)',
      position: 'relative',
      background: theme.palette.background.paper
    }
  };
};
var anthraciteTableRow = function anthraciteTableRow() {
  return {
    root: {
      height: '36px'
    },
    head: {
      height: 'unset'
    }
  };
};
var anthraciteTablePagination = function anthraciteTablePagination() {
  return {
    root: {},
    select: {
      boxShadow: 'none'
    }
  };
};
export { anthraciteTable, anthraciteTableCell, anthraciteTableRow, anthraciteTablePagination };
