// @jahia/design-system-kit/esm/layouts/content/TwoColumnsContent.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { withStyles } from '@material-ui/core';
import { compose } from 'recompose';
var styles = function styles(theme) {
  return {
    root: {
      flex: '1 1 0%',
      display: 'flex',
      flexDirection: 'row',
      backgroundColor: theme.palette.ui.alpha,
      padding: theme.spacing.unit * 4 + 'px ' + theme.spacing.unit * 4 + 'px 0'
    },
    left: {
      flex: '7 1 0%',
      display: 'flex',
      flexDirection: 'column',
      paddingRight: theme.spacing.unit * 2 + 'px'
    },
    right: {
      flex: '5 1 0%',
      display: 'flex',
      flexDirection: 'column',
      paddingLeft: theme.spacing.unit * 2 + 'px'
    }
  };
};
export var TwoColumnsContent = function TwoColumnsContent(_ref) {
  var children = _ref.children,
    rightCol = _ref.rightCol,
    classes = _ref.classes;
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.left
  }, children), /*#__PURE__*/React.createElement("div", {
    className: classes.right
  }, rightCol));
};
TwoColumnsContent.propTypes = {
  children: PropTypes.element.isRequired,
  rightCol: PropTypes.element.isRequired,
  classes: PropTypes.object.isRequired
};
export default compose(withStyles(styles))(TwoColumnsContent);
