// @jahia/moonstone/dist/components/Input/BaseInput/ControlledBaseInput.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["value", "searchContext", "id", "role", "placeholder", "isDisabled", "isReadOnly", "className", "size", "icon", "isShowClearButton", "onClear", "onChange", "onBlur", "onFocus", "focusOnField"];
import React, { useEffect, useRef } from 'react';
import clsx from 'clsx';
import { Cancel } from "../../../icons";
import { Button } from "../..";
import "./BaseInput.css";
export var ControlledBaseInput = function ControlledBaseInput(_ref) {
  var _ref$value = _ref.value,
      value = _ref$value === void 0 ? '' : _ref$value,
      searchContext = _ref.searchContext,
      id = _ref.id,
      role = _ref.role,
      placeholder = _ref.placeholder,
      _ref$isDisabled = _ref.isDisabled,
      isDisabled = _ref$isDisabled === void 0 ? false : _ref$isDisabled,
      _ref$isReadOnly = _ref.isReadOnly,
      isReadOnly = _ref$isReadOnly === void 0 ? false : _ref$isReadOnly,
      className = _ref.className,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      icon = _ref.icon,
      isShowClearButton = _ref.isShowClearButton,
      onClear = _ref.onClear,
      onChange = _ref.onChange,
      onBlur = _ref.onBlur,
      onFocus = _ref.onFocus,
      _ref$focusOnField = _ref.focusOnField,
      focusOnField = _ref$focusOnField === void 0 ? false : _ref$focusOnField,
      props = _objectWithoutProperties(_ref, _excluded);

  var hasSearchContext = typeof searchContext !== 'undefined';
  var isFilled = value !== '';
  var inputRef = useRef(null);
  var classNameProps = clsx('moonstone-baseInput', {
    'moonstone-baseInput_big': size === 'big'
  }, className);
  useEffect(function () {
    if (focusOnField) {
      inputRef.current.focus({
        preventScroll: true
      });
    }
  }, [focusOnField]);

  if (isShowClearButton && !onClear) {
    onClear = function onClear() {
      inputRef.current.value = '';
      var inputEvent = new Event('change');
      inputRef.current.dispatchEvent(inputEvent);
      onChange(inputEvent);
    };
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classNameProps
  }, hasSearchContext && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(searchContext.type, _extends({}, searchContext.props, {
    variant: "ghost",
    size: "small",
    maxWidth: "100px",
    className: clsx(searchContext.props.className, 'moonstone-searchContext-element')
  }))), /*#__PURE__*/React.createElement("input", _extends({
    ref: inputRef,
    className: clsx('moonstone-baseInput-element', {
      'moonstone-baseInput-element_hasIcon': icon,
      'moonstone-baseInput-element-withContext': hasSearchContext,
      'moonstone-baseInput-element_hasClearButton': onClear
    }),
    type: "text",
    role: role,
    value: value,
    id: id,
    placeholder: placeholder,
    disabled: isDisabled,
    readOnly: isReadOnly,
    onChange: onChange,
    onBlur: onBlur,
    onFocus: onFocus
  }, props)), icon && /*#__PURE__*/React.createElement("div", {
    className: clsx('moonstone-baseInput_icon', 'flexRow_nowrap', 'alignCenter')
  }, /*#__PURE__*/React.createElement(icon.type, _extends({}, icon.props, {
    focusable: "false"
  }))), onClear && isFilled && !isDisabled && !isReadOnly && /*#__PURE__*/React.createElement(Button, {
    className: "moonstone-baseInput_clearButton flexRow_center alignCenter",
    variant: "ghost",
    icon: /*#__PURE__*/React.createElement(Cancel, null),
    "aria-label": "Reset",
    onClick: onClear
  }));
};
ControlledBaseInput.displayName = 'ControlledBaseInput';
