// @jahia/design-system-kit/js/components/Chip/Chip.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _capitalize2 = _interopRequireDefault(require("lodash/capitalize"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _classnames = _interopRequireDefault(require("classnames"));
var _PropTypesConstants = _interopRequireDefault(require("../PropTypesConstants"));
var _excluded = ["root", "icon", "deleteIcon"],
  _excluded2 = ["variant", "color", "size", "classes"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var styles = function styles(theme) {
  return {
    root: {
      borderRadius: '4px'
    },
    primary: {
      background: theme.palette.brand.beta,
      color: theme.palette.invert.beta
    },
    iconPrimary: {
      color: theme.palette.invert.beta
    },
    secondary: {
      background: 'transparent',
      color: theme.palette.font.alpha
    },
    iconSecondary: {
      color: theme.palette.font.alpha
    },
    colorInverted: {
      color: theme.palette.invert.beta
    },
    deleteIcon: {
      color: theme.palette.ui.omega
    },
    sizeNormal: {
      // Todo normal size CSS
    },
    sizeCompact: {
      // Todo compact size CSS
    }
  };
};
var getClasses = function getClasses(_ref) {
  var variant = _ref.variant,
    color = _ref.color,
    size = _ref.size,
    _ref$classes = _ref.classes,
    root = _ref$classes.root,
    icon = _ref$classes.icon,
    deleteIcon = _ref$classes.deleteIcon,
    myClasses = _objectWithoutProperties(_ref$classes, _excluded);
  return {
    root: (0, _classnames["default"])(root, myClasses[variant], myClasses['color' + (0, _capitalize2["default"])(color)], myClasses['size' + (0, _capitalize2["default"])(size)]),
    icon: (0, _classnames["default"])(icon, myClasses['icon' + (0, _capitalize2["default"])(variant)]),
    deleteIcon: deleteIcon
  };
};
var Chip = (0, _core.withStyles)(styles, {
  name: 'DsChip'
})(function (_ref2) {
  var variant = _ref2.variant,
    color = _ref2.color,
    size = _ref2.size,
    classes = _ref2.classes,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/_react["default"].createElement(_core.Chip, _extends({
    classes: getClasses({
      variant: variant,
      color: color,
      size: size,
      classes: classes
    })
  }, props));
});
Chip.propTypes = process.env.NODE_ENV !== 'production' ? {
  size: _PropTypesConstants["default"].ChipSizes,
  color: _PropTypesConstants["default"].ChipColors,
  variant: _PropTypesConstants["default"].ChipVariants
} : {};
Chip.defaultProps = {
  color: 'default',
  variant: 'primary',
  size: 'normal'
};
Chip.displayName = 'DsChip';
var _default = Chip;
exports["default"] = _default;
