// @jahia/react-material/js/picker/PickerViewMaterialMultiple.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PickerViewMaterialMultiple = void 0;
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _icons = require("@material-ui/icons");
var _propTypes = _interopRequireDefault(require("prop-types"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var PickerViewMaterialMultiple = function PickerViewMaterialMultiple(props) {
  var _this = this;
  return /*#__PURE__*/_react["default"].createElement(_core.List, null, props.pickerEntries.map(function (entry) {
    return /*#__PURE__*/_react["default"].createElement(_core.ListItem, {
      key: entry.path,
      button: true,
      onClick: function onClick() {
        return entry.selectable ? props.onSelectItem(entry.path, !entry.selected, true) : props.onOpenItem(entry.path, !entry.open);
      }
    }, /*#__PURE__*/_react["default"].createElement(_core.ListItemIcon, {
      style: entry.selectable ? {} : {
        opacity: 0
      }
    }, entry.selected ? /*#__PURE__*/_react["default"].createElement(_icons.CheckBox, null) : /*#__PURE__*/_react["default"].createElement(_icons.CheckBoxOutlineBlank, null)), /*#__PURE__*/_react["default"].createElement(_core.ListItemText, {
      inset: true,
      primary: props.textRenderer ? props.textRenderer.call(_this, entry) : entry.name,
      style: {
        paddingLeft: entry.depth * props.theme.spacing.unit
      }
    }), /*#__PURE__*/_react["default"].createElement(_core.ListItemSecondaryAction, null, entry.openable && entry.hasChildren ? /*#__PURE__*/_react["default"].createElement(_core.IconButton, {
      onClick: function onClick() {
        return props.onOpenItem(entry.path, !entry.open);
      }
    }, entry.open ? /*#__PURE__*/_react["default"].createElement(_icons.ExpandLess, null) : /*#__PURE__*/_react["default"].createElement(_icons.ExpandMore, null)) : null));
  }));
};
exports.PickerViewMaterialMultiple = PickerViewMaterialMultiple;
PickerViewMaterialMultiple.defaultProps = {
  onSelectItem: function onSelectItem() {},
  onOpenItem: function onOpenItem() {},
  textRenderer: function textRenderer() {}
};
PickerViewMaterialMultiple.propTypes = {
  pickerEntries: _propTypes["default"].array.isRequired,
  onSelectItem: _propTypes["default"].func,
  onOpenItem: _propTypes["default"].func,
  textRenderer: _propTypes["default"].func,
  theme: _propTypes["default"].object.isRequired
};
exports.PickerViewMaterialMultiple = PickerViewMaterialMultiple = (0, _core.withTheme)()(PickerViewMaterialMultiple);
