// @jahia/react-material/js/picker/PickerViewMaterial.jsx
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PickerViewMaterial = void 0;
var _react = _interopRequireDefault(require("react"));
var _core = require("@material-ui/core");
var _icons = require("@material-ui/icons");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _iconRenderer = _interopRequireDefault(require("./iconRenderer"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var styles = function styles() {
  return {
    root: {
      position: 'relative'
    },
    loading: {
      opacity: 0.8
    },
    listItemSelected: {
      background: '#00a0e3',
      color: 'whitesmoke',
      '&:hover': {
        background: '#00a0e3',
        color: 'whitesmoke'
      }
    },
    listItem: {
      padding: '5px 10px 5px 0'
    },
    listItemLabel: {
      padding: '0px',
      '& h3': {
        fontSize: '0.875rem',
        color: '#676767',
        fontWeight: '100'
      }
    },
    listItemToggle: {
      marginRight: '10px',
      borderRadius: '0',
      width: 'auto'
    },
    listItemNodeTypeIcon: {
      marginRight: '5px'
    },
    selectedText: {
      color: 'whitesmoke!important'
    },
    loadingContainer: {
      position: 'absolute',
      width: '100%',
      height: '100%',
      zIndex: 999
    },
    toggleUnSelected: {
      color: '#00a0e3'
    },
    toggleSelected: {
      color: 'whitesmoke'
    },
    buttonContainer: {
      '&:hover': {
        backgroundColor: 'transparent'
      }
    }
  };
};
var PickerViewMaterial = function PickerViewMaterial(props) {
  var _this = this;
  var classes = props.classes,
    pickerEntries = props.pickerEntries,
    onOpenItem = props.onOpenItem,
    onSelectItem = props.onSelectItem,
    textRenderer = props.textRenderer,
    iconRenderer = props.iconRenderer,
    loading = props.loading;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.root
  }, loading && /*#__PURE__*/_react["default"].createElement("div", {
    className: classes.loadingContainer
  }), /*#__PURE__*/_react["default"].createElement(_core.List, {
    disablePadding: true,
    classes: {
      root: loading ? classes.root + ' ' + classes.loading : classes.root
    }
  }, pickerEntries.map(function (entry) {
    return /*#__PURE__*/_react["default"].createElement(_core.ListItem, {
      key: entry.path,
      button: true,
      divider: true,
      "data-jrm-role": "picker-item",
      className: entry.selected ? classes.listItem + ' ' + classes.listItemSelected : classes.listItem,
      onClick: function onClick() {
        return entry.selectable ? onSelectItem(entry.path, !entry.selected) : onOpenItem(entry.path, !entry.open);
      }
    }, /*#__PURE__*/_react["default"].createElement(_core.ListItemIcon, {
      className: entry.selected ? classes.listItemToggle + ' ' + classes.selectedText : classes.listItemToggle,
      style: {
        paddingLeft: (entry.depth + 1) * 20,
        opacity: entry.openable && entry.hasChildren ? 1 : 0
      }
    }, /*#__PURE__*/_react["default"].createElement(_core.IconButton, {
      className: classes.buttonContainer,
      disabled: !(entry.openable && entry.hasChildren),
      "data-jrm-role": "picker-item-toggle",
      "data-jrm-state": entry.open ? 'open' : 'closed',
      onClick: function onClick(event) {
        onOpenItem(entry.path, !entry.open);
        event.stopPropagation();
      }
    }, entry.open ? /*#__PURE__*/_react["default"].createElement(_icons.KeyboardArrowDown, {
      className: entry.selected ? classes.toggleSelected : classes.toggleUnSelected
    }) : /*#__PURE__*/_react["default"].createElement(_icons.KeyboardArrowRight, {
      className: entry.selected ? classes.toggleSelected : classes.toggleUnSelected
    }))), /*#__PURE__*/_react["default"].createElement(_core.ListItemIcon, {
      className: entry.selected ? classes.listItemNodeTypeIcon + ' ' + classes.selectedText : classes.listItemNodeTypeIcon
    }, iconRenderer ? iconRenderer.call(_this, entry) : _iconRenderer["default"].call(_this, entry)), /*#__PURE__*/_react["default"].createElement(_core.ListItemText, {
      inset: true,
      classes: entry.selected ? {
        root: classes.listItemLabel,
        primary: classes.selectedText
      } : {
        root: classes.listItemLabel
      },
      primary: textRenderer ? textRenderer.call(_this, entry) : entry.name,
      primaryTypographyProps: {
        'data-jrm-role': 'picker-item-text'
      }
    }));
  })));
};
exports.PickerViewMaterial = PickerViewMaterial;
PickerViewMaterial.defaultProps = {
  onSelectItem: function onSelectItem() {},
  onOpenItem: function onOpenItem() {},
  textRenderer: function textRenderer() {},
  iconRenderer: null
};
PickerViewMaterial.propTypes = {
  pickerEntries: _propTypes["default"].array.isRequired,
  onSelectItem: _propTypes["default"].func,
  onOpenItem: _propTypes["default"].func,
  textRenderer: _propTypes["default"].func,
  iconRenderer: _propTypes["default"].func,
  // eslint-disable-next-line react/boolean-prop-naming
  loading: _propTypes["default"].bool.isRequired,
  classes: _propTypes["default"].object.isRequired
};
exports.PickerViewMaterial = PickerViewMaterial = (0, _core.withTheme)()((0, _core.withStyles)(styles, {
  name: 'DxPickerViewMaterial'
})(PickerViewMaterial));
