// @jahia/react-material/js/theme/anthracite/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anthraciteLightTheme = exports.anthraciteDarkTheme = void 0;
var _merge2 = _interopRequireDefault(require("lodash/merge"));
var _core = require("@material-ui/core");
var _config = require("../common-base/config");
var _shape = require("./shape");
var _typography = require("./typography");
var _appbar = require("./overrides/appbar");
var _button = require("./overrides/button");
var _card = require("./overrides/card");
var _dialog = require("./overrides/dialog");
var _checkbox = require("./overrides/checkbox");
var _input = require("./overrides/input");
var _menu = require("./overrides/menu");
var _paper = require("./overrides/paper");
var _select = require("./overrides/select");
var _table = require("./overrides/table");
var _tabs = require("./overrides/tabs");
var _toolbar = require("./overrides/toolbar");
var _list = require("./overrides/list");
var _typography2 = require("./overrides/typography");
var _form = require("./overrides/form");
var _panel = require("./overrides/panel");
var _chip = require("./overrides/chip");
var _palette = require("./light/palette");
var _palette2 = require("./dark/palette");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
// Overrides

// Import of Light Anthracite palette

// Import of Dark Anthracite palette

var anthraciteLightThemeConfig = {
  palette: _palette.anthraciteLightPalette,
  shape: _shape.shape,
  typography: _typography.typography
};
var anthraciteLightThemeOverrides = function anthraciteLightThemeOverrides(theme) {
  return {
    MuiAppBar: (0, _appbar.anthraciteAppBar)(theme),
    MuiButton: (0, _button.anthraciteButton)(theme),
    MuiButtonBase: (0, _button.anthraciteButtonBase)(theme),
    MuiIconButton: (0, _button.anthraciteIconButton)(theme),
    MuiCard: (0, _card.anthraciteCard)(theme),
    MuiCardContent: (0, _card.anthraciteCardContent)(theme),
    MuiCardHeader: (0, _card.anthraciteCardHeader)(theme),
    MuiCheckbox: (0, _checkbox.anthraciteCheckbox)(theme),
    MuiChip: (0, _chip.anthraciteChip)(theme),
    MuiDialogTitle: (0, _dialog.anthraciteDialogTitle)(theme),
    MuiDialog: (0, _dialog.anthraciteDialog)(theme),
    MuiDialogActions: (0, _dialog.anthraciteDialogActions)(theme),
    MuiDialogContent: (0, _dialog.anthraciteDialogContent)(theme),
    MuiFormLabel: (0, _form.anthraciteFormLabel)(theme),
    MuiFormControlLabel: (0, _form.anthraciteFormControlLabel)(theme),
    MuiInput: (0, _input.anthraciteInput)(theme),
    MuiListItem: (0, _list.anthraciteListItem)(theme),
    MuiListItemSecondaryAction: _list.anthraciteListItemSecondaryAction,
    MuiListSubheader: _list.anthraciteListSubheader,
    MuiListItemText: _list.anthraciteListItemText,
    MuiMenu: (0, _menu.anthraciteMenu)(theme),
    MuiMenuItem: (0, _menu.anthraciteMenuItem)(theme),
    MuiExpansionPanel: (0, _panel.anthracitePanel)(theme),
    MuiExpansionPanelDetails: _panel.anthracitePanelDetails,
    MuiExpansionPanelActions: (0, _panel.anthracitePanelActions)(theme),
    MuiPaper: (0, _paper.anthracitePaper)(theme),
    MuiSelect: (0, _select.anthraciteSelect)(theme),
    MuiTable: (0, _table.anthraciteTable)(theme),
    MuiTableRow: (0, _table.anthraciteTableRow)(theme),
    MuiTablePagination: (0, _table.anthraciteTablePagination)(theme),
    MuiTableCell: (0, _table.anthraciteTableCell)(theme),
    MuiTab: (0, _tabs.anthraciteTab)(theme),
    MuiTabs: (0, _tabs.anthraciteTabs)(theme),
    MuiToolbar: (0, _toolbar.anthraciteToolBar)(theme),
    MuiTypography: (0, _typography2.anthraciteTypography)(theme)
  };
};
var anthraciteLightTheme = (0, _core.createMuiTheme)((0, _merge2["default"])({}, _config.commonBaseThemeConfig, anthraciteLightThemeConfig));
exports.anthraciteLightTheme = anthraciteLightTheme;
(0, _merge2["default"])(anthraciteLightTheme, {
  overrides: anthraciteLightThemeOverrides(anthraciteLightTheme)
});
var anthraciteDarkThemeConfig = {
  palette: _palette2.anthraciteDarkPalette,
  shape: _shape.shape,
  typography: _typography.typography
};
var anthraciteDarkThemeOverrides = function anthraciteDarkThemeOverrides(theme) {
  return {
    MuiAppBar: (0, _appbar.anthraciteAppBar)(theme),
    MuiButton: (0, _button.anthraciteButton)(theme),
    MuiButtonBase: (0, _button.anthraciteButtonBase)(theme),
    MuiIconButton: (0, _button.anthraciteIconButton)(theme),
    MuiCard: (0, _card.anthraciteCard)(theme),
    MuiCardContent: (0, _card.anthraciteCardContent)(theme),
    MuiCardHeader: (0, _card.anthraciteCardHeader)(theme),
    MuiCheckbox: (0, _checkbox.anthraciteCheckbox)(theme),
    MuiChip: (0, _chip.anthraciteChip)(theme),
    MuiDialogTitle: (0, _dialog.anthraciteDialogTitle)(theme),
    MuiDialog: (0, _dialog.anthraciteDialog)(theme),
    MuiDialogActions: (0, _dialog.anthraciteDialogActions)(theme),
    MuiDialogContent: (0, _dialog.anthraciteDialogContent)(theme),
    MuiFormLabel: (0, _form.anthraciteFormLabel)(theme),
    MuiFormControlLabel: (0, _form.anthraciteFormControlLabel)(theme),
    MuiInput: (0, _input.anthraciteInput)(theme),
    MuiListItem: (0, _list.anthraciteListItem)(theme),
    MuiListItemIcon: _list.anthraciteListItemIcon,
    MuiListItemSecondaryAction: _list.anthraciteListItemSecondaryAction,
    MuiListSubheader: _list.anthraciteListSubheader,
    MuiListItemText: _list.anthraciteListItemText,
    MuiMenu: (0, _menu.anthraciteMenu)(theme),
    MuiMenuItem: (0, _menu.anthraciteMenuItem)(theme),
    MuiExpansionPanel: (0, _panel.anthracitePanel)(theme),
    MuiExpansionPanelDetails: _panel.anthracitePanelDetails,
    MuiExpansionPanelActions: (0, _panel.anthracitePanelActions)(theme),
    MuiPaper: (0, _paper.anthracitePaper)(theme),
    MuiSelect: (0, _select.anthraciteSelect)(theme),
    MuiTable: (0, _table.anthraciteTable)(theme),
    MuiTableRow: (0, _table.anthraciteTableRow)(theme),
    MuiTableCell: (0, _table.anthraciteTableCell)(theme),
    MuiTab: (0, _tabs.anthraciteTab)(theme),
    MuiTabs: (0, _tabs.anthraciteTabs)(theme),
    MuiToolbar: (0, _toolbar.anthraciteToolBar)(theme),
    MuiTypography: (0, _typography2.anthraciteTypography)(theme)
  };
};
var anthraciteDarkTheme = (0, _core.createMuiTheme)((0, _merge2["default"])({}, _config.commonBaseThemeConfig, anthraciteDarkThemeConfig));
exports.anthraciteDarkTheme = anthraciteDarkTheme;
(0, _merge2["default"])(anthraciteDarkTheme, {
  overrides: anthraciteDarkThemeOverrides(anthraciteDarkTheme)
});
