// @jahia/design-system-kit/esm/components/LeftNavigation/BurgerMenuButton.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Tooltip, withStyles } from '@material-ui/core';
import { compose } from 'recompose';
import classNames from 'classnames';
var styles = function styles() {
  return {
    menuButton: {
      background: 'url(' + contextJsParameters.contextPath + '/engines/jahia-anthracite/images/logos/dx_logo_solid-white.png) center/100% no-repeat',
      width: '60px',
      height: '60px',
      backgroundSize: '100%'
    },
    menuButtonBlue: {
      background: 'url(' + contextJsParameters.contextPath + '/engines/jahia-anthracite/images/dx_logo_solid.png) center/100% no-repeat'
    }
  };
};
function openMenu() {
  var clickEvent = window.top.document.createEvent('MouseEvents');
  clickEvent.initEvent('click', true, true);
  window.top.document.getElementsByClassName('editmode-managers-menu')[0].dispatchEvent(clickEvent);
}
export var BurgerMenuButton = function BurgerMenuButton(_ref) {
  var classes = _ref.classes,
    isDrawerOpen = _ref.isDrawerOpen,
    title = _ref.title;
  if (isDrawerOpen) {
    return /*#__PURE__*/React.createElement("div", {
      className: classNames(classes.menuButton, classes.menuButtonBlue),
      "data-sel-role": "burger-menu",
      onClick: openMenu
    });
  }
  return /*#__PURE__*/React.createElement(Tooltip, {
    title: title,
    placement: "right"
  }, /*#__PURE__*/React.createElement("div", {
    className: classNames(classes.menuButton),
    "data-sel-role": "burger-menu",
    onClick: openMenu
  }));
};
BurgerMenuButton.propTypes = {
  isDrawerOpen: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired
};
BurgerMenuButton.defaultProps = {
  isDrawerOpen: false
};
export default compose(withStyles(styles, {
  name: 'DxBurgerMenuButton'
}))(BurgerMenuButton);
