// @jahia/design-system-kit/esm/components/LeftNavigation/LeftMenuItem.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Badge, Button, Typography, withStyles } from '@material-ui/core';
import { compose } from 'recompose';
import { toIconComponent } from '@jahia/react-material';
var styles = function styles(theme) {
  return {
    badgeRoot: {
      display: 'block'
    },
    badgeBadge: {
      right: 0,
      top: '-8px'
    },
    listItem: {
      margin: 0,
      height: '64px',
      display: 'block',
      width: '60px!important',
      paddingBottom: '10px!important'
    },
    typographyIconLight: {
      color: theme.palette.text.contrastText,
      fontSize: '9px',
      textTransform: 'uppercase',
      transition: 'all 0.2s ease-in 0s'
    },
    typographyIcon: {
      color: theme.palette.text.secondary,
      fontSize: '9px',
      textTransform: 'uppercase',
      transition: 'all 0.2s ease-in 0s'
    },
    colorClosed: {
      fill: theme.palette.text.contrastText,
      '& [fill="backgroundColor"]': {
        fill: theme.palette.layout.dark
      }
    },
    colorOpen: {
      fill: theme.palette.text.secondary,
      '& [fill="backgroundColor"]': {
        fill: theme.palette.background.paper
      }
    }
  };
};
export var LeftMenuItem = function LeftMenuItem(_ref) {
  var classes = _ref.classes,
    drawer = _ref.drawer,
    context = _ref.context;
  var _onClick = context.onClick,
    buttonLabel = context.buttonLabel,
    buttonIcon = context.buttonIcon,
    badge = context.badge;
  var icon = toIconComponent(buttonIcon, drawer ? {
    classes: {
      root: classes.colorOpen
    }
  } : {
    classes: {
      root: classes.colorClosed
    }
  });
  var Content = /*#__PURE__*/React.createElement(React.Fragment, null, Boolean(icon) && icon, /*#__PURE__*/React.createElement(Typography, {
    className: drawer ? classes.typographyIcon : classes.typographyIconLight,
    "data-sel-role": "left-menu-item-text"
  }, buttonLabel));
  return /*#__PURE__*/React.createElement(Button, {
    className: classes.listItem,
    onClick: function onClick(e) {
      return _onClick(context, e);
    }
  }, badge ? /*#__PURE__*/React.createElement(Badge, {
    badgeContent: badge,
    color: "error",
    classes: {
      root: classes.badgeRoot,
      badge: classes.badgeBadge
    },
    "data-sel-role": "badge"
  }, Content) : Content);
};
LeftMenuItem.propTypes = {
  context: PropTypes.object.isRequired,
  // eslint-disable-next-line react/boolean-prop-naming
  drawer: PropTypes.bool.isRequired,
  classes: PropTypes.object.isRequired
};
export default compose(withStyles(styles, {
  withTheme: true
}))(LeftMenuItem);
