// @jahia/design-system-kit/esm/components/LeftNavigation/LeftNavigation.jsx
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
import PropTypes from 'prop-types';
import React from 'react';
import { List, withStyles } from '@material-ui/core';
import classNames from 'classnames';
import { DisplayActions } from '@jahia/react-material';
import LeftMenuItem from './LeftMenuItem';
import { compose } from 'recompose';
import styleConstants from '../../theme/styleConstants';
var styles = function styles(theme) {
  return {
    root: {
      minWidth: styleConstants.screenMargin + styleConstants.leftNavigationWidth + 'px',
      paddingLeft: theme.spacing.unit * 4,
      display: 'flex',
      flexDirection: 'column'
    },
    rootOpenDrawer: {
      zIndex: theme.zIndex.modal,
      background: theme.palette.background.paper,
      overflow: 'visible !important',
      // Safari compatibility
      '-webkit-transform-style': 'preserve-3d' // Safari compatibility
    },

    rootClosedDrawer: {
      background: theme.palette.layout.dark,
      overflow: 'hidden'
    },
    spacer: {
      flex: '1 1 0%'
    },
    menuBurger: {
      paddingRight: '0px !important',
      paddingTop: '34px !important',
      paddingLeft: '0px !important',
      paddingBottom: '22px !important',
      boxShadow: 'none !important',
      background: 'none!important',
      transition: 'none'
    },
    drawerPaper: {
      background: theme.palette.background.paper,
      position: 'absolute',
      boxShadow: '2px 0 1px -2px rgba(0, 0, 21, 0.29)',
      left: styleConstants.screenMargin + styleConstants.leftNavigationWidth,
      width: styleConstants.leftNavigationDrawerWidth
    },
    drawerPaperClose: {
      width: 0,
      overflowX: 'hidden'
    },
    list: {},
    listBottom: {},
    drawerTree: {
      marginTop: '18px'
    }
  };
};
export var LeftNavigation = function LeftNavigation(_ref) {
  var _classNames;
  var context = _ref.context,
    classes = _ref.classes,
    actionsTarget = _ref.actionsTarget,
    secondaryActionsTarget = _ref.secondaryActionsTarget,
    drawer = _ref.drawer;
  var actionContext = _objectSpread(_objectSpread({}, context), {}, {
    drawer: drawer
  });
  return /*#__PURE__*/React.createElement("div", {
    className: classNames(classes.root, (_classNames = {}, _defineProperty(_classNames, classes.rootOpenDrawer, drawer.drawerOpen), _defineProperty(_classNames, classes.rootClosedDrawer, !drawer.drawerOpen), _classNames))
  }, /*#__PURE__*/React.createElement(List, {
    className: classes.list,
    component: "nav"
  }, /*#__PURE__*/React.createElement(DisplayActions, {
    target: actionsTarget,
    context: actionContext,
    render: function render(_ref2) {
      var context = _ref2.context;
      return /*#__PURE__*/React.createElement(LeftMenuItem, {
        context: context,
        drawer: drawer.drawerOpen
      });
    }
  })), /*#__PURE__*/React.createElement("div", {
    className: classes.spacer
  }), /*#__PURE__*/React.createElement(List, {
    className: classes.listBottom,
    component: "nav"
  }, /*#__PURE__*/React.createElement(DisplayActions, {
    target: secondaryActionsTarget,
    context: actionContext,
    render: function render(_ref3) {
      var context = _ref3.context;
      return /*#__PURE__*/React.createElement(LeftMenuItem, {
        context: context,
        drawer: drawer.drawerOpen
      });
    }
  })));
};
LeftNavigation.propTypes = {
  actionsTarget: PropTypes.string.isRequired,
  secondaryActionsTarget: PropTypes.string.isRequired,
  context: PropTypes.object.isRequired,
  drawer: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired
};
export default compose(withStyles(styles, {
  withTheme: true
}))(LeftNavigation);
