// @jahia/react-material/js/theme/legacy/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.legacyTheme = void 0;
var _merge2 = _interopRequireDefault(require("lodash/merge"));
var _core = require("@material-ui/core");
var _typography = require("./overrides/typography");
var _button = require("./overrides/button");
var _checkbox = require("./overrides/checkbox");
var _collapse = require("./overrides/collapse");
var _dialog = require("./overrides/dialog");
var _form = require("./overrides/form");
var _input = require("./overrides/input");
var _list = require("./overrides/list");
var _menu = require("./overrides/menu");
var _paper = require("./overrides/paper");
var _select = require("./overrides/select");
var _switch = require("./overrides/switch");
var _table = require("./overrides/table");
var _config = require("../common-base/config");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var legacyThemeConfig = {
  overrides: {
    MuiButton: _button.legacyButton,
    MuiIconButton: _button.legacyIconButton,
    MuiCheckbox: _checkbox.legacyCheckbox,
    MuiCollapse: _collapse.legacyCollapse,
    MuiDialogTitle: _dialog.legacyDialogTitle,
    MuiDialogContent: _dialog.legacyDialogContent,
    MuiDialogContentText: _dialog.legacyDialogContentText,
    MuiDialogActions: _dialog.legacyDialogActions,
    MuiFormControl: _form.legacyFormControl,
    MuiFormHelperText: _form.legacyFormHelperText,
    MuiFormControlLabel: _form.legacyFormControlLabel,
    MuiInput: _input.legacyInput,
    MuiListItemText: _list.legacyListItemText,
    MuiMenuItem: _menu.legacyMenuItem,
    MuiPaper: _paper.legacyPaper,
    MuiSelect: _select.legacySelect,
    MuiSwitch: _switch.legacySwitch,
    MuiTableRow: _table.legacyTableRow,
    MuiTableCell: _table.legacyTableCell,
    MuiTypography: _typography.legacyTypography
  }
};
var legacyTheme = (0, _core.createMuiTheme)((0, _merge2["default"])({}, _config.commonBaseThemeConfig, legacyThemeConfig));
exports.legacyTheme = legacyTheme;
